### PR TITLE
Migrate X-PACK Filebeat pipeline to main Beat pipeline

### PR DESCRIPTION
**Pull request by alexsapran**
## Proposed commit message

Migrating the x-pack filebeat pipeline into the centralized Beats pipelines

## Testing

- Simple build "default" case 
https://buildkite.com/elastic/beats/builds/5731#018ee6f4-2d18-4a97-882b-2aa0d90dfa65

- Adding all the labels
https://buildkite.com/elastic/beats/builds/5764

- Running cloud tests build 
https://buildkite.com/elastic/beats-xpack-filebeat/builds/1655#018eeb54-b690-431f-8d4b-59cd33a9870f
Same build from Jenkins https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats/detail/PR-38941/16/pipeline/2687/

Signed-off-by: Alexandros Sapranidis <alexandros@elastic.co>

### CODE REVIEW

**Comment by alexsapran** [comment]
There are some failures, which I re-triggered, they are known flaky windows tests.

File: .buildkite/x-pack/pipeline.xpack.filebeat.yml [hunk 1]
```diff
@@ -18,62 +18,311 @@ env:
...
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "x-pack/filebeat/build/*.xml"
+          - "x-pack/filebeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "x-pack/filebeat: Windows 11 Unit Tests"
+
+  - group: "x-pack/filebeat Extended Tests"
+    key: "x-pack-filebeat-extended-tests"
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.filebeat.yml:195]: shouldn't this group be skipped for non pull requests? in Jenkins it is part of the extended group https://github.com/elastic/beats/blob/main/x-pack/filebeat/Jenkinsfile.yml

File: .buildkite/x-pack/pipeline.xpack.filebeat.yml [hunk 2]
```diff
@@ -18,62 +18,311 @@ env:
...
+    steps:
+      - label: ":mac: MacOS x86_64 Unit Tests"
+        if: build.env("BUILDKITE_PULL_REQUEST") == "false"  || build.env("GITHUB_PR_LABELS") =~ /.*(macOS|aws).*/
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.filebeat.yml:198]: if we enable `if: build.env("BUILDKITE_PULL_REQUEST") == "false"` at the group level we can remove it from here

File: .buildkite/x-pack/pipeline.xpack.filebeat.yml [hunk 3]
```diff
@@ -18,62 +18,311 @@ env:
...
+        artifact_paths:
+          - "x-pack/filebeat/build/*.xml"
+          - "x-pack/filebeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "x-pack/filebeat: MacOS x86_64 Unit Tests"
+
+      - label: ":mac: MacOS arm64 Unit Tests"
+        skip: "https://github.com/elastic/beats/issues/33036"
+        if: build.env("BUILDKITE_PULL_REQUEST") == "false"  || build.env("GITHUB_PR_LABELS") =~ /.*(macOS|aws).*/
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.filebeat.yml:215]: if we enable `if: build.env("BUILDKITE_PULL_REQUEST") == "false"` at the group level we can remove it from here

**Review by dliappis — COMMENTED** [review]
Mostly LGTM left a few questions

File: .buildkite/x-pack/pipeline.xpack.filebeat.yml (hunk 1 above)
Comment (alexsapran) [reply .buildkite/x-pack/pipeline.xpack.filebeat.yml:195]: It is done, but the exception is in the steps not in the group.

This is how it was done with Filebeat.

File: .buildkite/x-pack/pipeline.xpack.filebeat.yml (hunk 2 above)
Comment (alexsapran) [reply .buildkite/x-pack/pipeline.xpack.filebeat.yml:198]: this is how we had it for Filebeat as well.

**Review by dliappis — APPROVED** [review]
LGTM

**Pull request merged**